main.js

```diff
@@ -2599,64 +2605,71 @@ function installAdapters() {
             }
         }
         else {
-            installArgs.push('install');
+            installArgs.push(commandScope);
             installArgs.push(name);
         }
-        logger.info(hostLogPrefix + ' ' + tools.appName + ' ' + installArgs.join(' ') + ' using ' + ((procs[task.id].downloadRetry < 3 && task.installedFrom) ? 'installedFrom' : 'installedVersion'));
+        logger.info(hostLogPrefix + ' ' + tools.appName + ' ' + installArgs.join(' ') + (task.rebuild ? '' : ' using ' + ((procs[task.id].downloadRetry < 3 && task.installedFrom) ? 'installedFrom' : 'installedVersion')));
         installArgs.unshift(__dirname + '/' + tools.appName + '.js');
 
         try {
             const child = spawn('node', installArgs, {windowsHide: true});
             if (child.stdout) {
                 child.stdout.on('data', data => {
                     data = data.toString().replace(/\n/g, '');
-                    logger.info(hostLogPrefix + ' ' + tools.appName + ' npm-install: ' + data);
+                    logger.info(hostLogPrefix + ' ' + tools.appName + ' npm-' + commandScope + ': ' + data);
                 });
             }
             if (child.stderr) {
                 child.stderr.on('data', data => {
                     data = data.toString().replace(/\n/g, '');
-                    logger.error(hostLogPrefix + ' ' + tools.appName + ' npm-install: ' + data);
+                    logger.error(hostLogPrefix + ' ' + tools.appName + ' npm-' + commandScope + ': ' + data);
                 });
             }
 
             child.on('exit', exitCode => {
-                logger.info(hostLogPrefix + ' ' + tools.appName + ' npm-install: exit ' + exitCode);
+                logger.info(hostLogPrefix + ' ' + tools.appName + ' npm-' + commandScope + ': exit ' + exitCode);
                 installQueue.shift();
                 if (exitCode === EXIT_CODES.CANNOT_INSTALL_NPM_PACKET) {
                     installQueue.push(task); // We add at the end again to try three times
-                } else if (!task.disabled) {
-                    if (!procs[task.id].config.common.enabled) {
-                        logger.info(hostLogPrefix + ' startInstance ' + task.id + ': should start instance but disabled, re-enable');
-                        states.setState(task.id + '.alive', {val: true, ack: false, from: hostObjectPrefix});
-                    }
-                    else {
-                        startInstance(task.id, task.wakeUp);
-                    }
                 } else {
-                    logger.debug(hostLogPrefix + ' ' + tools.appName + ' installation successfull but instance disabled');
+                    procs[task.id].needsRebuild = false;
+                    if (!task.disabled) {
+                        if (!procs[task.id].config.common.enabled) {
+                            logger.info(hostLogPrefix + ' startInstance ' + task.id + ': should start instance but disabled, re-enable');
```

Comment:
not sure what this is supposed to mean, maybe we can take this chance to update the message?

Reply:
In some cases when adapters are updated instances gets disabled temporarily ... this is the only reason I could think of that an instance got disabled

Reply:
Then maybe change the message to:
`... + ': instance is disabled but should be started, re-enabling it'`